hypothesis-python/src/hypothesis/core.py

```diff
@@ -776,7 +776,7 @@ def run(self):
                 ) % (
                     self.settings.timeout, runner.valid_examples),
                     self.settings)
-            if runner.valid_examples <= 1 and not (
+            if runner.valid_examples == 0 and not (
```

Comment:
If `runner.valid_examples == 0`, does the rest of this condition do anything at all?  From memory it doesn't, and we may also be able to get rid of `self.at_least_one_success.

Reply:
I think you're right. I'll take it out and see what happens.